src/Interpolation/InterpolateWithAdaptiveMinmod.cpp

```diff
@@ -0,0 +1,203 @@
+//
+//                   Case2: Interpolate ghost zone
+//                          i.e. InterpolateGhostZone() invokes InterpolateWithAdaptiveMinmod()
+//
+//                          2a. Interpolate conserved variables with original min-mod coefficient
+//
+//                              2b. If failed cell was found
+//                              --> Interpolate primitive variables with original min-mod coefficient
+//
+//                              2c. If failed cell was found again
+//                              --> Interpolate primitive variables with reducing min-mod coefficient
+//                                  until the min-mod coefficient is reduced to zero
+//
+//                4.  Interpolating primitive variables still preserves conservation because ghost zones do not
+//                    affect conservation.
+//
+// Parameter   :  CData           : Input coarse-grid array
+//                CSize           : Size of CData[]
+//                CStart          : (x,y,z) starting indices to perform interpolation on CData[]
+//                CRange          : Number of coarse cells along each direction to perform interpolation
+//                FData           : Output fine-grid array
+//                FSize           : Size of FData[]
+//                FStart          : (x,y,z) starting indices to store the interpolation results
+//                NComp           : Number of components in the CData and FData array
+//                IntScheme       : Interpolation scheme
+//                                  --> currently supported schemes include
+//                                      INT_MINMOD3D : MinMod-3D
+//                                      INT_MINMOD1D : MinMod-1D
+//                                      INT_VANLEER  : vanLeer
+//                                      INT_CQUAD    : conservative quadratic
+//                                      INT_QUAD     : quadratic
+//                                      INT_CQUAR    : conservative quartic
+//                                      INT_QUAR     : quartic
+//                UnwrapPhase     : Unwrap phase when OPT__INT_PHASE is on (for ELBDM only)
+//                Monotonic       : Ensure that all interpolation results are monotonic
+//                                  --> Useful for interpolating positive-definite variables, such as density, energy, ...
+//                OppSign0thOrder : Apply 0th-order interpolation if the values to be interpolated change
+//                                  signs in adjacent cells
+//                                  --> See Int_MinMod1D() for details
+//                IntGhostZone    : (true/false) --> (interpolate ghost zone/allocate new patches)
+//-------------------------------------------------------------------------------------------------------
+void InterpolateWithAdaptiveMinmod( real CData [], const int CSize[3], const int CStart[3], const int CRange[3],
```

Comment:
Integrate this function with `Interpolate()` first. I'll give more detailed comments after that.